src/Sarif.Viewer.VisualStudio/ResultTextMarker.cs

```diff
@@ -190,90 +193,114 @@ public void AddHighlightMarker(string highlightColor)
                 {
                     return false;
                 }
 
-                _tag = _tagger.AddTag(Region, tagSpan, _runIndex, new TextMarkerTag(Color));
+                this.tag = tagger.AddTag(this.mappedRegion, tagSpan, this.runIndex, new TextMarkerTag(Color));
             }
             else
             {
-                _tag = existingTag;
+                this.tag = existingTag;
             }
 
 
-            _tag.CaretEnteredTag += CaretEnteredTag;
-            _wpfTextView = wpfTextView;
-            _vsWindowFrame = vsWindowFrame;
-            _wpfTextView.Closed += this.TextViewClosed;
+            if (!SdkUIUtilities.TryGetActiveViewForTextBuffer(this.tag.DocumentPersistentSpan.Span.TextBuffer, out IWpfTextView wpfTextView))
+            {
+                return false;
+            }
+
+            this.tag.CaretEnteredTag += CaretEnteredTag;
+            wpfTextView.Closed += this.TextViewClosed;
 
             return true;
         }
 
         private void TextViewClosed(object sender, EventArgs e)
         {
-            _tag.CaretEnteredTag -= CaretEnteredTag;
+            if (sender is IWpfTextView wpfTextView)
+            {
+                wpfTextView.Closed -= this.TextViewClosed;
+            }
+
+            this.tag.CaretEnteredTag -= CaretEnteredTag;
         }
 
         // When the VS Editor tag has the caret moved inside of it, let's just pass along the region selection.
         private void CaretEnteredTag(object sender, EventArgs e) => this.RaiseRegionSelected?.Invoke(this, e);
 
-        private static bool TryCreateTextSpanWithinDocumentFromSourceRegion(Region region, IVsWindowFrame vsWindowFrame, out TextSpan textSpan)
+        private bool TryMapRegion()
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+
+            if (this.remapResult.HasValue)
+            {
+                return this.remapResult.Value;
+            }
+
+            if (string.IsNullOrEmpty(this.FullFilePath))
+            {
+                return false;
+            }
+
+            // Note: The call to TryRebaselineAllSarifErrors will ultimately
+            // set "this.FullFilePath" to the a new file path which is why calling
+            // File.Exists happens twice here.
+            if (!File.Exists(this.FullFilePath) && 
+                !CodeAnalysisResultManager.Instance.TryRebaselineAllSarifErrors(runIndex, this.UriBaseId, this.FullFilePath))
+            {
+                this.remapResult = false; 
+                return false;
+            }
+
+            if (File.Exists(this.FullFilePath) &&
+                Uri.TryCreate(this.FullFilePath, UriKind.Absolute, out Uri uri))
+            {
+                // Fill out the region's properties
+                FileRegionsCache regionsCache = CodeAnalysisResultManager.Instance.RunIndexToRunDataCache[runIndex].FileRegionsCache;
+                this.mappedRegion = regionsCache.PopulateTextRegionProperties(this.region, uri, true);
```

Comment:
>PopulateTextRegionProperties [](start = 49, length = 28)

Call this function twice and open a bug in the SDK if it is modified...

Reply:
Verified that calling this twice does not result in any bad behavior. I was confused because the incoming region is an insertion point and so the result is an insertion point.

Being an assertion point is corrected below to extend to the entire line.

---
In reply to: [489786511](https://github.com/microsoft/sarif-visualstudio-extension/pull/195#discussion_r489786511) [](ancestors = 489786511)